quality/transform_load_dlt.py

```diff
@@ -39,100 +39,98 @@ def quality_cci_bronze():
 
 # COMMAND ----------
 
-@dlt.table(name=f'quality_total_silver')
-def quality_total_silver():
-    bronze = dlt.read('quality_cci_bronze')
-    year_cols = list(col_name for col_name in bronze.columns if col_name.isnumeric())
-    return (bronze
-        .filter(F.col('category') == 'Spending: Total Expenditures')
-        .melt(ids=["country_name", "approved_or_executed"], 
-            values=year_cols, 
-            variableColumnName="year", 
-            valueColumnName="amount"
-        )
-        .filter(F.col('amount').isNotNull())
+def melt_and_pivot(df, id_vars=["country_name", "approved_or_executed"], groupby=["country_name", "year"]):
+    year_cols = list(col_name for col_name in df.columns if col_name.isnumeric())
+    melted = df.melt(
+        ids=id_vars,
+        values=year_cols,
+        variableColumnName="year",
+        valueColumnName="amount",
+    )
+    pivoted = melted.groupBy(groupby).agg(
+        F.sum(
+            F.when(F.col("approved_or_executed") == "Approved", F.col("amount"))
+        ).alias("approved"),
+        F.sum(
+            F.when(F.col("approved_or_executed") == "Executed", F.col("amount"))
+        ).alias("executed"),
```

Comment:
The old code filters out the null amounts while the new code doesn't. Would this result in many empty values for future years?